File: src/types/product.ts
export interface Product {
  id: number;
  platform_id: string;
  platform: string;
  title: string;
  description?: string;
  price: number;
  currency: string;
  location?: string;
  images: string[];
  url: string;
  created_at: string;
  updated_at: string;
}

export interface ProductFilters {
  platform?: string;
  search?: string;
  min_price?: number;
  max_price?: number;
  sort_by?: string;
  sort_order?: 'asc' | 'desc';
  page?: number;
  limit?: number;
}

export interface ProductList {
  items: Product[];
  total: number;
}

File: backend/app/schemas/product.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional

class ProductBase(BaseModel):
    platform_id: str
    platform: str
    title: str
    description: Optional[str] = None
    price: float
    currency: str = "CHF"
    location: Optional[str] = None
    images: List[str]
    url: str

class ProductCreate(ProductBase):
    pass

class Product(ProductBase):
    id: int
    created_at: datetime
    updated_at: datetime
    user_id: int

    class Config:
        from_attributes = True

class ProductList(BaseModel):
    items: List[Product]
    total: int